pkg/engine2/operational_rule/resource_placer.go

```diff
@@ -219,12 +219,14 @@ func (ctx OperationalRuleContext) findNumConnectionsToTypeForAvailableResources(
 	for _, availableResource := range availableResources {
 		var err error
 		var connections []construct.ResourceId
+		// We will look to see what direct dependencies are already existing in the same direction as the rule
+		// if we dont only look at direct, we risk getting incorrect results if the resource can have non functional connections
```

Comment:
Might not be necessary for a comment, but just for my knowledge, what "incorrect results" did this have that warrants the change?

Reply:
subnets in AZs, sometimes with EFS since the mount target isnt functional we see multiple subnets upstream of both AZs even though their direct dep and property are right